Update docs for getting token for origin trial (#2885)
MicrosoftDocs/edge-developer: microsoft-edge/origin-trials/index.md

ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.date: 10/20/2023
---
# Use Origin Trials in Microsoft Edge

> You can use Origin Trials to try out experimental APIs on live sites for a limited period of time.  When using Origin Trials, users of Microsoft Edge that visit your site may run code that uses experimental APIs.  To access the experimental APIs on each user machine, you don't need to go to `edge://flags` and turn on feature flags.  Instead, you register the origin (for example, `https://www.contoso.com:443`) that you want to use the experimental feature on, and you'll then be sent a token that you add to the code of your website.

Reviewer: ```suggestion
You can use Origin Trials to try out experimental APIs on your own live site for a limited period of time.  When using Origin Trials, users of Microsoft Edge that visit your site may run code that uses experimental APIs.  Users don't need to go to `edge://flags` on their own device to access the experimental APIs that your site uses. Instead, you register the origin (for example, `https://www.contoso.com:443`) that you want to use the experimental feature on, and you'll then be sent a token that you add to the code of your website.
```

Author: applied

> You can use Origin Trials to try out experimental APIs on your own live site for a limited period of time.  When using Origin Trials, users of Microsoft Edge that visit your site might run code that uses experimental APIs.  To support the experimental APIs, users don't need to go to `edge://flags` on their own device to access the experimental APIs that your site uses.  Instead, you register the origin (for example, `https://www.contoso.com:443`) that you want to use the experimental feature on, and you'll then be sent a token that you add to the code of your website.<!-- todo: indicate whether the token is automatically sent immediately, or if there's a manual process that takes a day or so -->

Author: @amandabaker - todo: indicate whether the token is automatically sent immediately, or if there's a manual process that takes a day or so

Reviewer 2: It's currently a manual process, but I'm not sure what we want to declare as an SLA for getting the token to the developer. @Reezaali WDYT?

Author: The surrounding Markdown content is Outdated.  Latest draft, in Files Changed tab of PR:
```
Instead, you register the origin (for example, `https://www.contoso.com:443`) that you want to 
use the experimental feature on, and 
you'll then be sent a token that you add to the code of your website.
Sending the token is not automatic and immediate, but is a manual process 
that can take a number of days.
<!-- todo: indicate whether the token is automatically sent immediately, 
or if there's a manual process that takes a day or so -->
```

To see a list of the available origin trials and register your origin to give them a try, see [Microsoft Edge Origin Trials](https://microsoftedge.github.io/MSEdgeExplainers/origin-trials/).  To assess the suitability of using the experimental APIs on your website, check the minimum version requirements for Microsoft Edge, and the trial end date.

You can also provide feedback to browser engineers and the web standards community about the design of the API, your use cases, or your experience using the APIs.

> For more information, see the **Getting Started** section in [Microsoft Edge Origin Trials](https://microsoftedge.github.io/MSEdgeExplainers/origin-trials/).

Reviewer 2: A lot of useful information about applying and opting out was removed from this page. Do you want to specify here that you can find more information about actually using the token (I'm not sure how to concisely word that). 
The relevant headings of content that was removed from here but can be found at the link provided are:
- Apply your token
- Opt out of an experiment
- Detect experimental features and provide a fallback

Author: @amandabaker @Reezaali -- this PR implements the strategy of no longer duplicating content in two different places:
* The article in this repo.
   * PR's condensed version: https://review.learn.microsoft.com/microsoft-edge/origin-trials/?branch=pr-en-us-2885
   * Previous/Live version: https://learn.microsoft.com/microsoft-edge/origin-trials/
* The green page https://microsoftedge.github.io/MSEdgeExplainers/origin-trials/, which contains detailed "Getting Started" & Dev Guidelines" sections.  Page outline (**bold** are equiv to your list of removed headings):

* Active Origin Trials
* Getting Started
   * Register for an Experiment
   * **Using an Origin Trial Token**
      * Option 1: Using a <meta> Tag
      * Option 2: Using an HTTP Header
* Developer Guidelines
   * Allowed Origins
   * Subdomains, URL Paths and Query Params
   * Origin Trial Duration
   * Providing Feedback on an Origin Trial
   * **Opting out of an experiment**
   * **Experimental feature implementation best practices**

To implement this redesign and confirm that strategy works, I first rewrote the article in this repo for clarity, and then removed the info that's in the Get Started section of the green page.

A previous commit in this PR shows my clarifying rewrite of the article in this repo - the result was close to the same wording as in that green page.  Possible plan: Move GetStart content from the green page into the present docs repo, such as restoring that initial commit.  Take best wording & clearest headings from that commit + from the green page.

The rendered PR page 
https://review.learn.microsoft.com/en-us/microsoft-edge/origin-trials/?branch=pr-en-us-2885
now reads as follows, replacing the previous content by a link:
```
For more information, see the Getting Started section in [Microsoft Edge Origin Trials](https://microsoftedge.github.io/MSEdgeExplainers/origin-trials/).
```

Author: @amandabaker - if you want, I can restore the improved/updated content in this article, per my earlier commits:

Here's a [commit](https://github.com/MicrosoftDocs/edge-developer/pull/2885/commits/c987bb1b7fd4eaf1352479c308cdd907ba0bf9b4) URL and subsequent [commit](https://github.com/MicrosoftDocs/edge-developer/pull/2885/commits/8c0ef7d8befee48d25182e21e41c19b33b90187c) URL from when I initially rewrote the learn.microsoft.com page for clarity, in this PR.

Reviewer 2: Sorry, my initial comment was worded poorly. I agree completely with consolidating, and the only change I intended to suggest was providing a hint at what info exists within the link you provided.

I like the update to point to the "Developer Guidelines" section since it hints at the content better than the "Getting Started" section did, so it LGTM.

When I initially reviewed this, I expected there to only be info about generating the OT at that link, so that's what triggered the original comment.

Author: result: keep PR as-is, with clarified link to green page 2 sections
